src/dapp/components/ui/HalveningBanner.tsx

```diff
@@ -1,24 +1,57 @@
+import "./Banner.css";
+
+import { useService } from "@xstate/react";
 import React, { useState } from "react";
 
-import alert from "../../images/ui/expression_alerted.png";
 import closeIcon from "../../images/ui/close.png";
-
-import "./Banner.css";
+import alert from "../../images/ui/expression_alerted.png";
+import { service } from "../../machine";
+import { numberWithBreaks } from "../../utils/number";
+import { getNextHalvingThreshold } from "../../utils/supply";
 
 export const Banner: React.FC = () => {
   const [show, setShow] = useState(true);
+  const [totalSupply, setTotalSupply] = React.useState<number>(0);
+
+  const [machineState, send] = useService<
+    Context,
+    BlockchainEvent,
+    BlockchainState
+  >(service);
+
+  React.useEffect(() => {
+    const load = async () => {
+      const notReady =
+        machineState.matches("loading") ||
+        machineState.matches("initial") ||
+        machineState.matches("registering") ||
+        machineState.matches("creating") ||
+        machineState.matches("onboarding");
+
+      if (notReady && totalSupply < 1) setTimeout(() => load(), 100);
+
+      const supply =
+        await service.machine.context.blockChain.totalSupply();
+
+      setTotalSupply(supply);
+    };
+
+    load();
+  }, [machineState]);
+
+  if (!show || !totalSupply) return null;
 
-  if (!show) {
-    return null;
-  }
+  console.log("totalSupply : ", totalSupply);
```

Comment:
this should be removed I think

Reply:
Good catch @zinkkrysty – thanks! Resolved!